### PR TITLE
Format docstring with ruff

### PR DESCRIPTION
**Pull request by beroda**
### Overview

Guess what ... ruff can ~make coffee :coffee:~ format docstrings https://docs.astral.sh/ruff/formatter/#docstring-formatting

Let's try out ...

Close #6467

### CODE REVIEW

File: pyvista/_plot.py [hunk 1]
```diff
@@ -200,9 +200,7 @@ def plot(
     ImageData. Note ``volume=True`` is passed.
 
     >>> import numpy as np
-    >>> grid = pv.ImageData(
-    ...     dimensions=(32, 32, 32), spacing=(0.5, 0.5, 0.5)
-    ... )
+    >>> grid = pv.ImageData(dimensions=(32, 32, 32), spacing=(0.5, 0.5, 0.5))
```
Comment (beroda) [inline pyvista/_plot.py:203]: Lines are wrapped more consistently than with blackdoc (length is 69 < limit of 75)

**Comment by user27182** [comment]
+1 for using `ruff` generally.
See also https://github.com/pyvista/pyvista/pull/6467#issuecomment-2272536196
There were previously concerns with `ruff` not formatting `rst` and `md` files

**Comment by user27182** [comment]
The ruff [configuration](https://docs.astral.sh/ruff/configuration/) states:

> Enable auto-formatting of code examples in docstrings. Markdown,
reStructuredText code/literal blocks and doctests are all supported.

Maybe I'm missing something but if markdown is supported, shouldn't it format this code block to use single quotes?
https://github.com/pyvista/pyvista/blob/6e59ba2115416e82c5994fce5abb6c5be271bbfc/CONTRIBUTING.rst?plain=1#L430-L437

But I'm not seeing `CONTRIBUTING.md` in the diff for this PR... There are also no `rst` files listed either, not sure if there are any codeblocks anywhere that _should_ be picked up, but still... makes me think `rst` files aren't covered either. 

Also, the current `blackdoc` config seems not to be working? Because if I try running `blackdoc CONTRIBUTING.rst`, this _does_ seem to work, but it errors out (it tries formatting bash code blocks and can't). But this clearly isn't the case with our current blackdoc config.


So, I'm +1 with this PR and dropping blackdoc and using ruff. But we should still figure out how to fill the gap and format code in `rst` and `md` files

**Comment by beroda** [comment]
> The ruff [configuration](https://docs.astral.sh/ruff/configuration/) states:
> 
> > Enable auto-formatting of code examples in docstrings. Markdown,
> > reStructuredText code/literal blocks and doctests are all supported.
> 
> Maybe I'm missing something but if markdown is supported, shouldn't it format this code block to use single quotes?

I agree that the ruff documentation is a bit misleading on this topic. I think they mean that ruff can format `Markdown` and `rst` python code blocks that are **inside the docstring of functions**. The examples in the configuration speak for themselves.
https://docs.astral.sh/ruff/settings/#format_docstring-code-format

> So, I'm +1 with this PR and dropping blackdoc and using ruff. But we should still figure out how to fill the gap and format code in `rst` and `md` files

I guess this gap will be filled once https://github.com/astral-sh/ruff/issues/8237 and https://github.com/astral-sh/ruff/issues/3792 are solved.

**Comment by beroda** [comment]
This PR should be merged after https://github.com/pyvista/pyvista/pull/6994 to be compatible.
Move to draft.

File: pyproject.toml [hunk 2]
```diff
@@ -175,6 +175,7 @@ pyvista = ['py.typed']
 # reading the docstrings in text terminals.
 line-length = 75
 
+
```
Comment (tkoyama010) [inline pyproject.toml:178]:
```suggestion
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**